korge/src/common/korlibs/korge/ui/UIButton.kt

```diff
@@ -99,6 +99,16 @@ open class UIButton(
             setInitialState()
         }
 
+    override var enabled: Boolean
+        get() = super.enabled
+        set(value) {
+            val valueChanged = value != mouseEnabled
+            mouseEnabled = value
+
+            if (valueChanged)
+                updatedUIButton(enable = value)
+        }
```

Comment:
Why not updating UIView from:

```suggestion
open var enabled: Boolean
    get() = mouseEnabled
    set(value) {
        mouseEnabled = value
        updateState()
    }
```

to

```suggestion
open var enabled: Boolean
    get() = mouseEnabled
    set(value) {
        if (mouseEnabled == value) return
        mouseEnabled = value
        updateState()
    }
```

? so we don't need overrigint this? wouldn't that work?

Since `updateState` is calling `updatedUIButton`, and since mouseEnabled has been set, when updatedUIButton gets the enabled property is already set to the right value, right?

Reply:
@soywiz I would have prefered it that way as well, but we need that to be able to supply the `enable` argument for `updatedUIButton`.

If you have a nicer idea to find out, if `enabled` changed since the last update, I would be happy to see that :D

Reply:
And I liked your version for checking if the last value is the same within the setter method, so I changed it to yours :)

Reply:
Okay. Let's keep it as this is 👍